src/core/core-varlink.c

```diff
@@ -202,11 +203,23 @@ static int vl_method_subscribe_managed_oom_cgroups(
                 void *userdata) {
 
         _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+        _cleanup_free_ char *comm = NULL;
         Manager *m = ASSERT_PTR(userdata);
+        pid_t pid;
         int r;
 
         assert(link);
 
+        r = varlink_get_peer_pid(link, &pid);
+        if (r < 0)
+                return r;
+
+        (void) get_process_comm(pid, &comm);
+
+        /* This is meant to be a deterrent and not actual security. */
+        if (!streq(comm, "systemd-oomd"))
```

Comment:
Isn't it better to check which unit the process belongs to? Still not really a security measure, but a bit more robust given we don't allow duplicate units